protocol/x/prices/keeper/market_test.go

```diff
@@ -170,15 +170,29 @@ func TestCreateMarket_Errors(t *testing.T) {
 			exchangeConfigJson: validExchangeConfigJson,
 			expectedErr: errorsmod.Wrap(
 				types.ErrInvalidInput,
-				"market param 0 exponent -6 does not match market price 0 exponent -5",
+				"market param 1 exponent -6 does not match market price 1 exponent -5",
+			).Error(),
+		},
+		"Pair already exists": {
+			pair:               "0-0",
+			minExchanges:       uint32(2),
+			minPriceChangePpm:  uint32(50),
+			price:              constants.FiveBillion,
+			exchangeConfigJson: validExchangeConfigJson,
+			expectedErr: errorsmod.Wrap(
+				types.ErrMarketParamPairAlreadyExists,
+				"0-0",
 			).Error(),
 		},
 	}
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			ctx, keeper, _, _, _, _ := keepertest.PricesKeepers(t)
+			ctx, keeper, _, _, _, mockTimeKeeper := keepertest.PricesKeepers(t)
 			ctx = ctx.WithTxBytes(constants.TestTxBytes)
 
+			mockTimeKeeper.On("Now").Return(constants.TimeT)
```

Comment:
Why do we need a timekeeper now?

Reply:
The other test cases never succesfully create a market param. They all fail before hitting this line https://github.com/dydxprotocol/v4-chain/blob/ab14ab8b82c13c08afa1226f51a945ad3f00093c/protocol/x/prices/keeper/market.go#L64
The additional test case relies on a market param already existing so it has to be able to succeed once.